### PR TITLE
feat: added comment section for listings

### PR DESCRIPTION
**Pull request by VictorCodebase**
# For issue #297  

## changes made 
- Created UI to display user comments below listing details
- Implemented a dynamic connection between number of comments and total reviews
- Implemented a feature populating sample comments based on the star rating

## preview video:

https://github.com/Azure-Samples/contoso-real-estate/assets/135356007/66aac896-78ba-4e31-aa0d-a0c7b6fd1275

# screenshots:
image
Screenshot 2023-08-29 142057

## feedback request
I am more than excited to submit this PR! Please feel more than free to point out areas I can improve😃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Enterprise-grade Reference Architecture for JavaScript
 
+
```
Comment (anfibiacreativa) [inline README.md:3]: Please remove this additional breaks introduced in this PR. Thank you!

File: packages/api/data/template/listings.json [hunk 2]
```diff
@@ -293,4 +293,4 @@
...
 {"title":"Luxurious apartment by the park","city":"Houston","bedrooms":2,"bathrooms":1,"amenities":["fireplace","heating","garden"],"description":"Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur.","img":"pic-pink.png","slug":{"$numberLong":"608072"}}
 {"title":"Bright apartment close to the park","city":"Sidney","bedrooms":3,"bathrooms":3,"amenities":["balcony","elevator","sauna"],"description":"Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus.","img":"pic-yellowgreen.png","slug":{"$numberLong":"586853"}}
-{"title":"Cozy apartment in the heart of the city","city":"Beijing","bedrooms":5,"bathrooms":2,"amenities":["fireplace","heating","wi-fi"],"description":"Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur.","img":"pic-pink.png","slug":{"$numberLong":"629909"}}
```
Comment (anfibiacreativa) [inline packages/api/data/template/listings.json:296]: I'm not really sure about the intended change in this file. Can you please give us some pointers? Thank you!

File: packages/portal/src/app/rentalpage/rentalpage.component.html [hunk 3]
```diff
@@ -1,61 +1,41 @@
 <section class="listing-container">
   <section class="listing-photos">
     <main>
-      <div
-        [ngClass]="{ 'loading-background': isLoading() }"
-        [ngStyle]="{ 'background-image': 'url(/assets/images/' + listing().photos[0] + ')' }"
-        alt="Main listing photo"
-      ></div>
+      <div [ngClass]="{ 'loading-background': isLoading() }"
```
Comment (anfibiacreativa) [inline packages/portal/src/app/rentalpage/rentalpage.component.html:4]: Please review formatting. It got somehow changed. Thanks!

File: packages/portal/src/app/rentalpage/rentalpage.component.html [hunk 4]
```diff
@@ -1,61 +1,41 @@
...
-        [ngClass]="{ 'loading-background': isLoading() }"
-        [ngStyle]="{ 'background-image': 'url(/assets/images/' + listing().photos[2] + ')' }"
-        alt="Listing photo #2"
-      ></div>
-      <div
-        [ngClass]="{ 'loading-background': isLoading() }"
-        [ngStyle]="{ 'background-image': 'url(/assets/images/' + listing().photos[3] + ')' }"
-        alt="Listing photo #3"
-      >
+      <div [ngClass]="{ 'loading-background': isLoading() }"
```
Comment (anfibiacreativa) [inline packages/portal/src/app/rentalpage/rentalpage.component.html:9]: Same here. There appear to be a lot of changes that are just the formatting of the document having unintentionally changed...

**Review by anfibiacreativa — CHANGES_REQUESTED** [review]
Hi! Thank you for your contribution. There appear to be a lot of changes that are only the formatting affected. There are also some changes to environment variables files that are difficult to understand and a lot of comments added to the files that I'm not sure you intended to push? Please make sure to revert formatting changes and remove comments so we can identify the intended changes! Thank you, again!

**Comment by VictorCodebase** [comment]
I have removed all those unnecessary edits😅. I think I accidentally made them trying to familiarize myself with the code.

Im not sure; should I open a new pull request or how do I go about this?

File: packages/api/data/template/listings.json (hunk 2 above)
Comment (VictorCodebase) [reply packages/api/data/template/listings.json:296]: I was trying to see where the ratings were coming from😂. No change was intended in this file

The line 296 seems identical to mine. Ive tried deleting line 297 but its still being highlighted in red.

I literally compared both lines - they seem identical

image

**Review by sinedied — CHANGES_REQUESTED** [review]
Could you run `npm run format` at the root of the repo, so all your code is properly formatted? Thanks

**Comment by anfibiacreativa** [comment]
> @sinedied I have run npm run format and gosh it has modified a whole lot of files. Should I try reverting or is it fine.
> 
> I have removed the comments about playwright also

Hi @VictorCodebase You will need to pull from upstream to fix conflicts and also make sure only the intended changes necessary to accomplish the feature, are part of the PR delta. 
I will label it `pending review` so you can take your time to push your changes.

**Comment by VictorCodebase** [comment]
I have managed to resolve all conflicts. Nust in time to run out of codespaces😭😭

Screenshot 2023-09-01 135428

**Comment by VictorCodebase** [comment]
Hi @anfibiacreativa ; Could I provide a gif of the code running (like the website running with no errors) instead of going through all the altered files reverting them🥲

**Comment by manekinekko** [comment]
Hi @VictorCodebase I'd recommend you close this PR and create a new clean PR with intended changes only.

**Comment by VictorCodebase** [comment]
Thanks so much @manekinekko .

**Pull request closed without merging**